File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    var answer = 0;

    let next = n;
    let i = 1;
    while(1){
        let count_n = 0;
        let count_next = 0;
        next = n + i;
        
        let bin_n = n.toString(2);
        let bin_next = next.toString(2);
        
        for(num_n of bin_n){
            if(num_n == 1){
                count_n += 1;
            }
        }
        for(num_next of bin_next){
            if(num_next == 1){
                count_next += 1;
            }
        }
        if(count_n == count_next){
            answer = next;
            break;
        }
        i++;
    }
    return answer;
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = n + 1;
        
        while(true){
            if(isCheck(n, answer)){ return answer;}
            answer += 1;
        }
    }
    
    public boolean isCheck (int n, int next){
        if (toBin(n) == toBin(next)) return true;
        return false;
    }
    
    public int toBin(int num){
        int countOne = 0;
        
        while(num > 0){
            int r = num % 2;
            if (r == 1){
                countOne += 1;
            }
            num /= 2;
        }
        return countOne;
    }
}